packages/object/src/index.ts

```diff
@@ -102,23 +102,35 @@ export class DRPObject implements IDRPObject {
 	}
 
 	// This function is black magic, it allows us to intercept calls to the DRP object
-	proxyDRPHandler(): ProxyHandler<object> {
+	proxyDRPHandler(parentProp?: string): ProxyHandler<object> {
 		const obj = this;
 		return {
 			get(target, propKey, receiver) {
-				if (typeof target[propKey as keyof object] === "function") {
+				const value = Reflect.get(target, propKey, receiver);
+
+				if (typeof value === "function") {
+					const fullPropKey = parentProp
+						? `${parentProp}.${String(propKey)}`
+						: String(propKey);
 					return new Proxy(target[propKey as keyof object], {
 						apply(applyTarget, thisArg, args) {
 							if ((thisArg.operations as string[]).includes(propKey as string))
-								obj.callFn(
-									propKey as string,
-									args.length === 1 ? args[0] : args,
-								);
+								obj.callFn(fullPropKey, args.length === 1 ? args[0] : args);
 							return Reflect.apply(applyTarget, thisArg, args);
 						},
 					});
 				}
-				return Reflect.get(target, propKey, receiver);
+
+				if (propKey === "acl" && typeof value === "object" && value !== null) {
```

Comment:
should check if value ==="object" first for better clarity.
```suggestion
				if (typeof value === "object" && value !== null && propKey === "acl") {
```